backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/builder/vminfo/VmInfoBuildUtils.java

```diff
@@ -1671,14 +1671,16 @@ public Long getNvdimmTotalSize(VM vm, MemoizingSupplier<Map<String, HostDevice>>
     }
 
     public static int maxNumberOfVcpus(VM vm) {
-        return FeatureSupported.supportedInConfig(ConfigValues.HotPlugCpuSupported, vm.getCompatibilityVersion(),
-                vm.getClusterArch()) && !VmCpuCountHelper.isResizeAndPinPolicy(vm) ? VmCpuCountHelper.calcMaxVCpu(vm, vm.getCompatibilityVersion())
-                        : VmCpuCountHelper.getDynamicNumOfCpu(vm);
+        return FeatureSupported.hotPlugCpu(vm.getCompatibilityVersion(), vm.getClusterArch(), vm.getCpuPinningPolicy()) ?
+                VmCpuCountHelper.calcMaxVCpu(vm, vm.getCompatibilityVersion())
+                : VmCpuCountHelper.getDynamicNumOfCpu(vm);
     }
 
     public static boolean isVmWithHighNumberOfX86Vcpus(VM vm) {
+        int numOfCpus = VmCpuCountHelper.isDynamicCpuTopologySet(vm) ||
+                vm.getCpuPinningPolicy() == CpuPinningPolicy.DEDICATED ?
+                VmCpuCountHelper.getDynamicNumOfCpu(vm) : maxNumberOfVcpus(vm);
```

Comment:
can't we make use of the changes in FeatureSupported and maxNumberOfVcpus to replace this logic?

Reply:
yeah, i think we can just call `maxNumberOfVcpus`.